README.md

```diff
@@ -1,2 +1,175 @@
+     */
+    @RequireRole(Role.Teacher)
+    fun updateAccountInfo(context: RequestContext, request: UpdateAccountInfoRequest): AccountInfoResponse
+
+    ...
+}
+```
+
+```protobuf
+message LoginRequest {
+  AuthenticationProvider provider = 1;
+  classtime.service.common.Role role = 2;
+}
+
+message LoginResponse {
+  string redirect_url = 1;
+}
+
+message UserProfileRequest {
+  string account_id = 1;
+}
+
+message UserProfileResponse {
+  string email = 1;
+  string first_name = 2;
+  string last_name = 3;
+  classtime.service.common.Role role = 4;
+}
+```
+
+## URLs construction
+
+Each API end point consists of 3 concatenated parts:
+1. Our server public host: https://www.classtime.com/service/public/
+2. Service name (e.g. Account, School, Library)
+3. Service method name (e.g. createAccount, login, associateTeacher)
+
+Based on [interface sample](./README.md#interface-sample), we could specify such endpoints:
+```
+https://www.classtime.com/service/public/Account/login
+https://www.classtime.com/service/public/Account/getPublicUserProfile
+```
+
+## Role permissions
+
+Each service and service method contains an annotation limiting access to specific roles. And of course, method annotations have higher priority. For example, in the [interface sample](./README.md#interface-sample), there are:
+- `AccountService` with `@Public` service annotation, which makes all methods available for all.
+- `login` method without an annotation. And as a result, it uses `AccountService` scope - `@Public`.
+- `getPublicUserProfile` method with `@RequireRole(Role.Teacher)`. Only teachers are allowed to call this endpoint.
+
+### Auth Header
+
+To get access scope of some roles, you need to have an access token tied to an account with that role.  
+Your access token must be sent in Auth header with a request in the following schema:
+
+```
+Authorization: JWT ${accessToken}
+```
+
+#### Get testing access token
+The easiest way to get an access token for some test purposes is through classtime website:
+1. Go to a [Classtime website](https://www.classtime.com/auth/login) and authorize using any method (e.g. with Google).
+2. Open "classtime.com" cookies in the developer console of the browser. (see e.g. [how to do it in Google Chrome](https://developer.chrome.com/docs/devtools/application/cookies/#open))
+3. Copy the value of the cookie with the name: "service-jwt-0". It is the access token of your teacher account.
+
+## JSON requests
+
+JSON requests are the easiest point to start. However, we are recommending to use binary requests.  
+Note: **All requests use the HTTP POST method**
+
+### Public AccountService.login call:
+```bash
+curl -v https://www.classtime.com/service/public/Account/login \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Accept: application/json,*/*' \
+    -d '{ provider: "Google", role: "Teacher" }'
+```
+```json
+Response: { "redirectUrl": "https://accounts.google.com..." }
```

Comment:
Better to put the Reponse: above the json, then the code block has a valid json syntax.